fix(css): add snappport glossary section (#36300)
mdn/content: files/en-us/glossary/scroll_snap/index.md

---
title: Scroll snap
slug: Glossary/Scroll_snap
page-type: glossary-definition
---

{{GlossarySidebar}}

> **Scroll snapping** is when content "snaps" to a specific position, rather than stopping at any random point, when a scroll operation finishes. Normal scrolling operations lack precision. They don't align to a paragraph, a sentence, or an image boundary. For example, in a carousel, a scrolling operation could finish in the middle of an image, leaving it partially visible. Web developers have long relied on JavaScript-based solutions to control scrolling and address this shortcoming default behavior.

Author: ```suggestion
**Scroll snapping** is when content "snaps" to a specific position, rather than stopping at any random point, when a scroll operation finishes. Normal scrolling operations lack precision. They don't align to a paragraph, a sentence, or an image boundary. For example, in a carousel, a scrolling operation could finish in the middle of an image, leaving it partially visible. Web developers have long relied on JavaScript-based solutions. Browsers have recently started supporting [CSS scroll-snapping features](/en-US/docs/Web/CSS/CSS_scroll_snap) that enable defining scroll-snap containers and snapping behaviors.
```

> Now CSS supports the scroll snap feature out of the box. The feature provides a well-controlled scrolling experience. Web developers can mark each [scroll container](/en-US/docs/Glossary/Scroll_container) with boundaries for scroll operations. The scroll operations then finish at these boundaries (snap positions). In the above carousel example, as the user finishes scrolling the carousel, its visible image will snap into place.

Author: ```suggestion
Scroll snapping is a well-controlled scrolling experience, whereby developers define an element as a [scroll container](/en-US/docs/Glossary/Scroll_container) with boundaries for scroll operations. Scroll operations then finish at these snap position boundaries, with the scrolled to content snapping into place. In the above carousel example, as the user finishes scrolling the carousel, its visible image will snap into place.
```

- `Scroll snap container`

  - : If a box has {{CSSXref("scroll-snap-type")}} set to a value other than `none`, then the box captures snap positions. A box's **scroll snap container** is a box's nearest snap-position capturing scroll container ancestor.

- `Snapport`

>   - : The **snapport** is the area of the [scrollport](/en-US/docs/Glossary/Scroll_container#scrollport) defined using {{CSSXref("scroll-padding")}} property set on the scroll container. When calculating snap positions, the area is used as the alignment container for the scroll snap areas. By default, it is the same as the scroll container's visual viewport. The [snap areas](#snap_area) are aligned to the scrollport.

Author: ```suggestion
- : The **snapport** ​is the area of the [scrollport](/en-US/docs/Glossary/Scroll_container#scrollport) that is used as the alignment container for the scroll snap areas when calculating snap positions. By default, it is the same as the scroll container's visual viewport, but is the area of the scrollport defined by the {{CSSXref("scroll-padding")}} property. The [snap areas](#snap_area) are aligned to the scrollport.
```

---
title: Scroll snap
slug: Glossary/Scroll_snap
page-type: glossary-definition
---

{{GlossarySidebar}}

Normal scrolling operations lack precision. They don't align to a paragraph, a sentence, or an image boundary. For example, in a carousel, a scrolling operation could finish in the middle of an image, leaving it partially visible. Web developers have long relied on JavaScript-based solutions to control scrolling and address this shortcoming.

> The CSS **Scroll Snap** feature provides a well-controlled scrolling experience. Web developers can mark each [scroll container](/en-US/docs/Glossary/Scroll_container) with boundaries for scroll operations. Scroll operations then finish at these boundaries (snap positions). In the above carousel example, as the user finishes scrolling the carousel, its visible image will snap into place.

Reviewer: We should define "scroll snap" as a user experience feature, not as a CSS feature. CSS merely enables such UX on the web, but you can implement it in other places too.

> ## Scroll snap container

Reviewer: I would prefer these short definitions to be a DL rather than H2s for glossary

If a box has {{CSSXref("scroll-snap-type")}} set to a value other than `none`, then the box captures snap positions. A box's **scroll snap container** is a box's nearest snap-position capturing scroll container ancestor.

## Snapport

> The **snapport** is the area of the scroll container defined using {{CSSXref("scroll-padding")}} property set on the scroll container. By default, it is the same as the scroll container's visual viewport. The [snap areas](#snap_area) are aligned to the scrollport.

Reviewer 2: > the scroll container's visual viewport

is this the same as the scrollport?

Author: yes
